cpp/examples/tpch/q1.cpp

```diff
@@ -0,0 +1,215 @@
+    auto tbl_view = table->view();
+    auto keys = cudf::table_view{{tbl_view.column(0), tbl_view.column(1)}};
+
+    auto quantity = tbl_view.column(2);
+    auto extendedprice = tbl_view.column(3);
+    auto discount = tbl_view.column(4);
+    auto discprice = tbl_view.column(8);
+    auto charge = tbl_view.column(9);
+
+    cudf::groupby::groupby groupby_obj(keys);
+    std::vector<cudf::groupby::aggregation_request> requests;
+
+    requests.emplace_back(cudf::groupby::aggregation_request());
+    requests[0].aggregations.push_back(cudf::make_sum_aggregation<cudf::groupby_aggregation>());
+    requests[0].aggregations.push_back(cudf::make_mean_aggregation<cudf::groupby_aggregation>());
+    requests[0].values = quantity;
+
+    requests.emplace_back(cudf::groupby::aggregation_request());
+    requests[1].aggregations.push_back(cudf::make_sum_aggregation<cudf::groupby_aggregation>());
+    requests[1].aggregations.push_back(cudf::make_mean_aggregation<cudf::groupby_aggregation>());
+    requests[1].values = extendedprice;
+
+    requests.emplace_back(cudf::groupby::aggregation_request());
+    requests[2].aggregations.push_back(cudf::make_mean_aggregation<cudf::groupby_aggregation>());
+    requests[2].values = discount;
+
+    requests.emplace_back(cudf::groupby::aggregation_request());
+    requests[3].aggregations.push_back(cudf::make_sum_aggregation<cudf::groupby_aggregation>());
+    requests[3].values = discprice;
+
+    requests.emplace_back(cudf::groupby::aggregation_request());
+    requests[4].aggregations.push_back(cudf::make_sum_aggregation<cudf::groupby_aggregation>());
+    requests[4].values = charge;
+
+    requests.emplace_back(cudf::groupby::aggregation_request());
+    requests[5].aggregations.push_back(cudf::make_count_aggregation<cudf::groupby_aggregation>());
+    requests[5].values = charge;
+
+    auto agg_results = groupby_obj.aggregate(requests);
+    auto result_key = std::move(agg_results.first);
+
+    auto returnflag = std::make_unique<cudf::column>(result_key->get_column(0));
+    auto linestatus = std::make_unique<cudf::column>(result_key->get_column(1));
+
+    std::vector<std::unique_ptr<cudf::column>> columns;
+    columns.push_back(std::move(returnflag));
+    columns.push_back(std::move(linestatus));
+    columns.push_back(std::move(agg_results.second[0].results[0]));
+    columns.push_back(std::move(agg_results.second[0].results[1]));
+    columns.push_back(std::move(agg_results.second[1].results[0]));
+    columns.push_back(std::move(agg_results.second[1].results[1]));
+    columns.push_back(std::move(agg_results.second[2].results[0]));
+    columns.push_back(std::move(agg_results.second[3].results[0]));
+    columns.push_back(std::move(agg_results.second[4].results[0]));
+    columns.push_back(std::move(agg_results.second[5].results[0]));
+    
+    return std::make_unique<cudf::table>(std::move(columns));
+}
+
+int main() {
+    auto s = std::chrono::high_resolution_clock::now();
```

Comment:
We have a light-weight `Timer` implemented for examples at: https://github.com/rapidsai/cudf/blob/branch-24.08/cpp/examples/parquet_io/parquet_io.hpp#L135

It might be worthwhile (if not too much overhead) to move it out of the `parquet_io.hpp` header to a common header file and use it for the `parquet_io` and this example as well reducing some current and future code efforts. 🙂

Reply:
Sure, working on this

Reply:
Added the `Timer` to the utils.hpp for now. Could we extract the timer code into the detail module ?